packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js

```diff
@@ -109,6 +116,109 @@ ModifierButton.propTypes = {
   }).isRequired,
 };
 
+const isBlockActive = (editor, value) => {
+  const { selection } = editor;
+
+  if (!selection) return false;
+
+  let matchCondition;
+
+  switch (value.type) {
+    case 'heading':
+      matchCondition = (n) => n.type === value.type && n.level === value?.level;
+      break;
+    default:
+      matchCondition = (n) => n.type === value.type;
+      break;
+  }
+
+  const match = Array.from(
+    Editor.nodes(editor, {
+      at: Editor.unhangRange(editor, selection),
+      match: (n) => !Editor.isEditor(n) && SlateElement.isElement(n) && matchCondition(n),
+    })
+  );
+
+  return match.length > 0;
+};
+
+const toggleBlock = (editor, value) => {
+  const { type, level } = value;
+
+  const newProperties = {
+    type,
+    level: level || null,
+  };
+
+  Transforms.setNodes(editor, newProperties);
+};
+
+const BlocksDropdown = () => {
+  const editor = useSlate();
+  const [optionSelected, setOptionSelected] = React.useState(Object.keys(blockItems)[0]);
+
+  /**
+   * @param {string} optionKey - key of the heading selected
+   */
+  const selectOption = (optionKey) => {
+    toggleBlock(editor, blockItems[optionKey].value);
+
+    setOptionSelected(optionKey);
+  };
+
+  return (
+    <Select
+      startIcon={<Icon as={blockItems[optionSelected].icon} />}
+      onChange={selectOption}
+      placeholder={blockItems[optionSelected].label}
+      value={optionSelected}
+    >
+      {Object.keys(blockItems).map((key) => (
+        <BlockOption
+          key={blockItems[key].name}
+          value={key}
+          label={blockItems[key].label}
+          icon={blockItems[key].icon}
+          onActive={setOptionSelected}
+          optionSelected={optionSelected}
+        />
+      ))}
+    </Select>
+  );
+};
+
+const BlockOption = ({ value, icon, label, onActive, optionSelected }) => {
+  const { formatMessage } = useIntl();
+  const editor = useSlate();
+
+  const isActive = isBlockActive(editor, blockItems[value].value);
+  const isSelected = value === optionSelected;
+
+  if (isActive && !isSelected) {
+    onActive(value);
+  }
```

Comment:
I think this needs to be in a useEffect to make sure we're in sync with isActive and isSelected instead on running on every render